src/Workspaces/CSharp/Portable/Utilities/SpeculationAnalyzer.cs

```diff
@@ -278,6 +278,16 @@ protected override bool ReplacementChangesSemanticsForNodeLanguageSpecific(Synta
             Debug.Assert(previousOriginalNode == null || previousOriginalNode.Parent == currentOriginalNode);
             Debug.Assert(previousReplacedNode == null || previousReplacedNode.Parent == currentReplacedNode);
 
+            if (currentOriginalNode.IsKind(SyntaxKind.CaseSwitchLabel, SyntaxKind.ConstantPattern))
+            {
+                var expression = (ExpressionSyntax)currentReplacedNode.ChildNodes().First();
+                if (expression.WalkDownParentheses().IsKind(SyntaxKind.DefaultLiteralExpression))
+                {
+                    // We can't have a default literal inside a case label or constant pattern.
+                    return true;
+                }
+            }
```

Comment:
@CyrusNajmabadi having the check here will be the most correct and will work for all IDE features, the only thing I don't like is that this is just a special edge case that is at the very beginning and steals focus from the rest of this method which is far more important

also, this doesn't really feel like a "semantic" change all that much

Reply:
But I absolutely feel like this code has to be somewhere in the speculation analyzer and not scattered around individual features. Doing this fixes real bugs in other features. One example I found is:
```c#
const bool b = default;

switch (true)
{
    case b:
        break;
}
```
do inline temporary on 'b' - it will use produce broken code. But with this change, it will introduce a cast and the resulting code will be correct:
```c#
switch (true)
{
    case ((bool)(default)):
        break;
}
```

Reply:
> also, this doesn't really feel like a "changing the semantics" all that much

"changes semantics" is just the term we used for "breaks the code" :)  

> the only thing I don't like is that this is just a special edge case that is at the very beginning and steals focus from the rest of this method which is far more important

The method is just a bunch of checks that was added in the order we found and fixed thigns while doing this feature.  I don't see any problem with your addition.